user_guide_src/source/extending/composer_packages.rst

```diff
@@ -0,0 +1,194 @@
+
+You can make the :doc:`../general/modules` you create into Composer packages,
+or can create a Composer package for CodeIgniter 4.
+
+.. contents::
+    :local:
+    :depth: 2
+
+****************
+Folder Structure
+****************
+
+Here's a typical directory structure for a Composer package::
+
+    your-package-name/
+    ├── .gitattributes
+    ├── .gitignore
+    ├── README.md
+    ├── composer.json
+    ├── src/
+    │   └── YourClass.php
+    └── tests/
+        └── YourClassTest.php
+
+**********************
+Creating composer.json
+**********************
+
+In the root of your package directory, create a **composer.json** file. This file
+defines metadata about your package and its dependencies.
+
+The ``composer init`` command helps you create it.
+
+For example, **composer.json** might look like this::
+
+    {
+        "name": "your-vendor/your-package",
+        "description": "Your package description",
+        "type": "library",
+        "license": "MIT",
+        "autoload": {
+            "psr-4": {
+                "YourVendor\\YourPackage\\": "src/"
+            }
+        },
+        "authors": [
+            {
+                "name": "Your Name",
+                "email": "yourname@example.com"
+            }
+        ],
+        "require": {
+            // Any dependencies required by your package go here
+        },
+        "require-dev": {
+            // Any development dependencies (e.g., PHPUnit) go here
+        }
+    }
+
+Package Name
+============
+
+The ``name`` field is important here. Package names are generally written in the
+format "vendor-name/package-name" with all lowercase. Here is a common example:
+
+- ``your-vendor-name``: The name that identifies the vendor (creator of the package),
+  such as the name of you or your organization.
+- ``your-package-name``: The name of the package you are creating.
+
+Thus, it is important to make the name unique and distinguish it from other packages.
+Uniqueness is especially important when publishing.
+
+Namespace
+=========
+
+The package name then determines the vendor namespace in ``autoload.psr4``.
+
+If your package name is ``your-vendor/your-package``, the vendor namespace must
+be ``YourVendor``. So you would write like the following::
+
+    "autoload": {
+        "psr-4": {
+            "YourVendor\\YourPackage\\": "src/"
+        }
+    },
+
+This setting instructs Composer to autoload the source code for your package.
+
+***************************
+Preparing Development Tools
+***************************
+
+There are many tools that help ensure quality code. So you should use them.
+You can easily install and configure such tools with
+`CodeIgniter DevKit <https://github.com/codeigniter4/devkit>`_.
+
+Installing DevKit
+=================
+
+In the root of your package directory, run the following commands::
+
+    > composer config minimum-stability dev
+    > composer config prefer-stable true
+    > composer require --dev codeigniter4/devkit
```

Comment:
The `>` can be removed so that the commands can be copy pasted.

Reply:
We now use `>` in all command samples in the user guide.
Removing it only here is not consistent.

Reply:
I sent #7757 and fixed these.